### ISSUE TITLE
Strict Secure Cookie policy error in Admob after updating to Android 11

### ISSUE THREAD

**Issue by newit-huytt** [googleads/googleads-mobile-flutter#158]
## Bug report
Native ad cause crash app on simulator Android, API 30 

**Describe the bug**
https://stackoverflow.com/questions/64605612/strict-secure-cookie-policy-error-in-admob-after-updating-to-android-11
### Steps to reproduce

1. Config native ad as codelab https://codelabs.developers.google.com/codelabs/admob-inline-ads-in-flutter#7 
2. Run app on Android simular with API 30

### Expected behavior

Firstly app run success but after 3s it was crash and throw the error:

```
Strict Secure Cookie policy does not allow setting a secure cookie for
http://googleads.g.doubleclick.net/ for apps targeting >= R. Please either use the 'https:' scheme for 
this URL or omit the 'Secure' directive in the cookie value.
```

**Comment by newit-huytt** [comment]
I found the reason for cause error. It came from my code

**Comment by mcradha** [comment]
Facing the same issue after I migrated from firebase_admob to google_mobile_ads plugin for flutter. My app doesn't crash but rewarded ads dont show.  I too raised a question on this [here](https://github.com/googleads/googleads-mobile-flutter/discussions/129):

**Comment by erickcchoi** [comment]
Exact same issue here, Hope them to fix the problem asap.

**Comment by Sebastien-Remy** [comment]
Exact same issue here, with last version 18 may 2021

**Comment by Sebastien-Remy** [comment]
> Facing the same issue after I migrated from firebase_admob to google_mobile_ads plugin for flutter. My app doesn't crash but rewarded ads dont show. I too raised a question on this [here](https://github.com/googleads/googleads-mobile-flutter/discussions/129):

your link return 404 issue

**Comment by amicharski** [comment]
Have the same problem with google_mobile_ads 0.13.0 and it's almost July

**Comment by dutch-game-studio** [comment]
We have the same issue

**Comment by wildsurfer** [comment]
Same issue

**Comment by michael-ottink** [comment]
Same issue

**Comment by maheshmnj** [comment]
Hi @newit-huytt , Thanks for filing the issue, I ran the [example app](https://github.com/googlecodelabs/admob-inline-ads-in-flutter/tree/main/complete_kotlin_swift) and could not reproduce the crash though, but the ad failed to load on an emulator but works fine on the real device also I couldn't see the reported error 

```console
Strict Secure Cookie policy does not allow setting a secure cookie for
http://googleads.g.doubleclick.net/ for apps targeting >= R. Please either use the 'https:' scheme for 
this URL or omit the 'Secure' directive in the cookie value.
```


<details>
<summary>Output</summary>

Emulator (Api 30) | Xiaomi k20 Pro (Andorid 11)
:-: | :-:
<video src='https://user-images.githubusercontent.com/31410839/131484103-b888471c-dd14-4fd1-aacd-fbb86b2636c5.mov' width=180/> | <video src='https://user-images.githubusercontent.com/31410839/131484122-00585b9b-866d-4548-a2f7-46a77f36ded0.mov' width=180/>


</details>



<details>
<summary>logs</summary>

```
Launching lib/main.dart on sdk gphone arm64 in debug mode...
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:50091/8G1OGQizI4s=/ws
I/WebViewFactory( 5051): Loading com.google.android.webview version 83.0.4103.106 (code 410410683)
I/.admobinlinead( 5051): The ClassLoaderContext is a special shared library.
D/nativeloader( 5051): classloader namespace configured for unbundled product apk. library_path=/product/app/WebViewGoogle/lib/arm64:/product/app/WebViewGoogle/WebViewGoogle.apk!/lib/arm64-v8a:/product/app/TrichromeLibrary/TrichromeLibrary.apk!/lib/arm64-v8a:/product/lib64:/system/product/lib64
I/.admobinlinead( 5051): The ClassLoaderContext is a special shared library.
D/nativeloader( 5051): classloader namespace configured for unbundled product apk. library_path=/product/app/WebViewGoogle/lib/arm64:/product/app/WebViewGoogle/WebViewGoogle.apk!/lib/arm64-v8a:/product/app/TrichromeLibrary/TrichromeLibrary.apk!/lib/arm64-v8a:/product/lib64:/system/product/lib64
I/cr_VariationsUtils( 5051): Requesting new seed from IVariationsSeedServer
I/cr_LibraryLoader( 5051): Loaded native library version number "83.0.4103.106"
I/cr_CachingUmaRecorder( 5051): Flushed 3 samples from 3 histograms.
E/chromium( 5051): [ERROR:filesystem_posix.cc(62)] mkdir /data/user/0/com.codelab.flutter.admobinlineads/cache/WebView/Crashpad: No such file or directory (2)
W/.admobinlinead( 5051): Accessing hidden method Landroid/media/AudioManager;->getOutputLatency(I)I (greylist, reflection, allowed)
D/HostConnection( 5051): HostConnection::get() New Host Connection established 0x7ac20d8050, tid 5178
D/HostConnection( 5051): HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buffer ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_async_frame_commands ANDROID_EMU_gles_max_version_3_0
D/EGL_emulation( 5051): eglCreateContext: 0x7ab21665a0: maj 3 min 0 rcv 3
D/EGL_emulation( 5051): eglMakeCurrent: 0x7ab21665a0: ver 3 0 (tinfo 0x7a52169e10) (first time)
W/cr_media( 5051): Requires BLUETOOTH permission
W/cr_MediaCodecUtil( 5051): HW encoder for video/avc is not available on this device.
D/EGL_emulation( 5051): eglCreateContext: 0x7ab2161740: maj 3 min 0 rcv 3
I/Ads     ( 5051): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7142D2739190028E1BC646C676069A53")) to get test ads on this device.
D/zzge    ( 5051): File /data/user/0/com.codelab.flutter.admobinlineads/cache/1613498354782.dex not found. No need for deletion
I/DynamiteModule( 5051): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:21400
I/DynamiteModule( 5051): Selected remote version of com.google.android.gms.ads.dynamite, version >= 21400
V/DynamiteModule( 5051): Dynamite loader version >= 2, using loadModule2NoCrashUtils
I/DynamiteModule( 5051): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:21400
I/DynamiteModule( 5051): Selected remote version of com.google.android.gms.ads.dynamite, version >= 21400
V/DynamiteModule( 5051): Dynamite loader version >= 2, using loadModule2NoCrashUtils
D/GASS    ( 5051): Cannot dynamite load clearcut
W/ConnectionTracker( 5051): Exception thrown while unbinding
W/ConnectionTracker( 5051): java.lang.IllegalArgumentException: Service not registered: lm@805b24d
W/ConnectionTracker( 5051): 	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1757)
W/ConnectionTracker( 5051): 	at android.app.ContextImpl.unbindService(ContextImpl.java:1874)
W/ConnectionTracker( 5051): 	at android.content.ContextWrapper.unbindService(ContextWrapper.java:792)
W/ConnectionTracker( 5051): 	at ce.b(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):1)
W/ConnectionTracker( 5051): 	at ce.a(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):5)
W/ConnectionTracker( 5051): 	at ln.A(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):10)
W/ConnectionTracker( 5051): 	at ky.a(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):3)
W/ConnectionTracker( 5051): 	at dy.run(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):2)
W/ConnectionTracker( 5051): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
W/ConnectionTracker( 5051): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/ConnectionTracker( 5051): 	at iw.run(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):15)
W/Ads     ( 5051): Not retrying to fetch app settings
I/Ads     ( 5051): Received log message: <Google:HTML> Incorrect native ad response. Click actions were not properly specified
I/Ads     ( 5051): Ad failed to load : 0
I/flutter ( 5051): Ad load failed (code=0 message=Internal error.)
I/Ads     ( 5051): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7142D2739190028E1BC646C676069A53")) to get test ads on this device.
I/Ads     ( 5051): Received log message: <Google:HTML> Incorrect native ad response. Click actions were not properly specified
I/Ads     ( 5051): Ad failed to load : 0
I/flutter ( 5051): Ad load failed (code=0 message=Internal error.)

```
</details>

**Comment by maheshmnj** [comment]
Can some one confirm if the crash is happening on a real Android device too running on Android 11?

Thanks

**Comment by maheshmnj** [comment]
Hi @newit-huytt , Thanks for filing the issue, I ran the [example app](https://github.com/googlecodelabs/admob-inline-ads-in-flutter/tree/main/complete_kotlin_swift) and could not reproduce the crash though, but the ad failed to load on an emulator but works fine on the real device also I couldn't see the reported error 

```console
Strict Secure Cookie policy does not allow setting a secure cookie for
http://googleads.g.doubleclick.net/ for apps targeting >= R. Please either use the 'https:' scheme for 
this URL or omit the 'Secure' directive in the cookie value.
```


<details>
<summary>Output</summary>

Emulator (Api 30) | Xiaomi k20 Pro (Andorid 11)
:-: | :-:
<video src='https://user-images.githubusercontent.com/31410839/131484103-b888471c-dd14-4fd1-aacd-fbb86b2636c5.mov' width=180/> | <video src='https://user-images.githubusercontent.com/31410839/131484122-00585b9b-866d-4548-a2f7-46a77f36ded0.mov' width=180/>


</details>



<details>
<summary>logs</summary>

```
Launching lib/main.dart on sdk gphone arm64 in debug mode...
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:50091/8G1OGQizI4s=/ws
I/WebViewFactory( 5051): Loading com.google.android.webview version 83.0.4103.106 (code 410410683)
I/.admobinlinead( 5051): The ClassLoaderContext is a special shared library.
D/nativeloader( 5051): classloader namespace configured for unbundled product apk. library_path=/product/app/WebViewGoogle/lib/arm64:/product/app/WebViewGoogle/WebViewGoogle.apk!/lib/arm64-v8a:/product/app/TrichromeLibrary/TrichromeLibrary.apk!/lib/arm64-v8a:/product/lib64:/system/product/lib64
I/.admobinlinead( 5051): The ClassLoaderContext is a special shared library.
D/nativeloader( 5051): classloader namespace configured for unbundled product apk. library_path=/product/app/WebViewGoogle/lib/arm64:/product/app/WebViewGoogle/WebViewGoogle.apk!/lib/arm64-v8a:/product/app/TrichromeLibrary/TrichromeLibrary.apk!/lib/arm64-v8a:/product/lib64:/system/product/lib64
I/cr_VariationsUtils( 5051): Requesting new seed from IVariationsSeedServer
I/cr_LibraryLoader( 5051): Loaded native library version number "83.0.4103.106"
I/cr_CachingUmaRecorder( 5051): Flushed 3 samples from 3 histograms.
E/chromium( 5051): [ERROR:filesystem_posix.cc(62)] mkdir /data/user/0/com.codelab.flutter.admobinlineads/cache/WebView/Crashpad: No such file or directory (2)
W/.admobinlinead( 5051): Accessing hidden method Landroid/media/AudioManager;->getOutputLatency(I)I (greylist, reflection, allowed)
D/HostConnection( 5051): HostConnection::get() New Host Connection established 0x7ac20d8050, tid 5178
D/HostConnection( 5051): HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buffer ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_async_frame_commands ANDROID_EMU_gles_max_version_3_0
D/EGL_emulation( 5051): eglCreateContext: 0x7ab21665a0: maj 3 min 0 rcv 3
D/EGL_emulation( 5051): eglMakeCurrent: 0x7ab21665a0: ver 3 0 (tinfo 0x7a52169e10) (first time)
W/cr_media( 5051): Requires BLUETOOTH permission
W/cr_MediaCodecUtil( 5051): HW encoder for video/avc is not available on this device.
D/EGL_emulation( 5051): eglCreateContext: 0x7ab2161740: maj 3 min 0 rcv 3
I/Ads     ( 5051): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7142D2739190028E1BC646C676069A53")) to get test ads on this device.
D/zzge    ( 5051): File /data/user/0/com.codelab.flutter.admobinlineads/cache/1613498354782.dex not found. No need for deletion
I/DynamiteModule( 5051): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:21400
I/DynamiteModule( 5051): Selected remote version of com.google.android.gms.ads.dynamite, version >= 21400
V/DynamiteModule( 5051): Dynamite loader version >= 2, using loadModule2NoCrashUtils
I/DynamiteModule( 5051): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:21400
I/DynamiteModule( 5051): Selected remote version of com.google.android.gms.ads.dynamite, version >= 21400
V/DynamiteModule( 5051): Dynamite loader version >= 2, using loadModule2NoCrashUtils
D/GASS    ( 5051): Cannot dynamite load clearcut
W/ConnectionTracker( 5051): Exception thrown while unbinding
W/ConnectionTracker( 5051): java.lang.IllegalArgumentException: Service not registered: lm@805b24d
W/ConnectionTracker( 5051): 	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1757)
W/ConnectionTracker( 5051): 	at android.app.ContextImpl.unbindService(ContextImpl.java:1874)
W/ConnectionTracker( 5051): 	at android.content.ContextWrapper.unbindService(ContextWrapper.java:792)
W/ConnectionTracker( 5051): 	at ce.b(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):1)
W/ConnectionTracker( 5051): 	at ce.a(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):5)
W/ConnectionTracker( 5051): 	at ln.A(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):10)
W/ConnectionTracker( 5051): 	at ky.a(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):3)
W/ConnectionTracker( 5051): 	at dy.run(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):2)
W/ConnectionTracker( 5051): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
W/ConnectionTracker( 5051): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/ConnectionTracker( 5051): 	at iw.run(:com.google.android.gms.dynamite_measurementdynamite@201817048@20.18.17 (040400-0):15)
W/Ads     ( 5051): Not retrying to fetch app settings
I/Ads     ( 5051): Received log message: <Google:HTML> Incorrect native ad response. Click actions were not properly specified
I/Ads     ( 5051): Ad failed to load : 0
I/flutter ( 5051): Ad load failed (code=0 message=Internal error.)
I/Ads     ( 5051): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7142D2739190028E1BC646C676069A53")) to get test ads on this device.
I/Ads     ( 5051): Received log message: <Google:HTML> Incorrect native ad response. Click actions were not properly specified
I/Ads     ( 5051): Ad failed to load : 0
I/flutter ( 5051): Ad load failed (code=0 message=Internal error.)

```
</details>

Can some one confirm if the crash is happening on a real Android device too running on Android 11?

Thanks

**Comment by rekire** [comment]
I run in the same situation, just that I don't use ads at all. I'm trying to set cookies for a webview in a flutter app. I think we have here some general issue with setting cookies to webviews with flutter targeting Android >= R.

**Comment by maheshmnj** [comment]
@rekire 

> I run in the same situation, just that I don't use ads at all. I'm trying to set cookies for a webview in a flutter app. I think we have here some general issue with setting cookies to webviews with flutter targeting Android >= R.

In that case, it doesn't seem to be ad specific issue and the official webView plugin doesn't yet support setting cookies feature you can follow the [feature request here.](https://github.com/flutter/flutter/issues/27597)

**Comment by rekire** [comment]
I found out that there is a not very oblivious way to get this working. Details are in the linked ticket above. I'm not sure if this actually helps to resolve the bug in this plugin, but you could check the code to find this out.

**Comment by maheshmnj** [comment]
Reopening since this is not yet fixed.

**Comment by solankiherry** [comment]
same issue :(

**Comment by alexandr-efimov** [comment]
Any info how to fix?

`
E/chromium(11868): [ERROR:cookie_manager.cc(137)] Strict Secure Cookie policy does not allow setting a secure cookie for http://googleads.g.doubleclick.net/ for apps targeting >= R. Please either use the 'https:' scheme for this URL or omit the 'Secure' directive in the cookie value.
`

**Comment by rekire** [comment]
As said earlier the cookies are set incorrectly, the origin need to be set.

If you have full access to the source code (I think just Googler have this) this should be easy to fix. Check the linked issues from me before.

**Comment by 0xZOne** [comment]
> Any info how to fix?
> 
> `E/chromium(11868): [ERROR:cookie_manager.cc(137)] Strict Secure Cookie policy does not allow setting a secure cookie for http://googleads.g.doubleclick.net/ for apps targeting >= R. Please either use the 'https:' scheme for this URL or omit the 'Secure' directive in the cookie value.`


See https://chromium-review.googlesource.com/c/chromium/src/+/2040685 for more details.

**Comment by dellarosamarco** [comment]
I resolved it by using "google_mobile_ads: ^0.13.0" insted of the latest version

**Comment by QFREY** [comment]
This is the solution 

Go to  android -> app -> src -> main -> AndroidManifest.xml

Paste this in the AndroidManifest.xml in the application tag
android:usesCleartextTraffic="true"
it should look something like this when you are done

        <application
        android:label="Your app name"
        android:name="${applicationName}"
        android:icon="@mipmap/ic_launcher"
       android:usesCleartextTraffic="true">

this will make no difference in the emulator if you want the emulator to work as well do the following

go to android -> app -> src -> debug-> AndroidManifest.xml

open the AndroidManifest.xml 
then you have to add this code below the manifest tag

<application android:usesCleartextTraffic="true"/>

Now the ads will work in the emulator as well.

**Comment by bdstph** [comment]
Same issue here

**Comment by NVentimiglia** [comment]
Thanks for the report, we will take a look.

**Comment by malandr2** [comment]
The Android Google Mobile Ads team has addressed this issue and it has been fixed in version 21.0.0. The GMA flutter plugin has version 21.0.0 from [version 2.0.0](https://pub.dev/packages/google_mobile_ads/changelog) and above. If you are running into this issue, please update your flutter plugin version to the latest available version to ensure the fix.
